modal/object.py

```diff
@@ -202,13 +202,20 @@ def is_hydrated(self) -> bool:
         """mdmd:hidden"""
         return self._is_hydrated
 
-    async def _try_hydrate(self) -> bool:
-        if not self._is_hydrated and self._hydrate_lazily:
+    async def resolve(self):
+        if self._is_hydrated:
+            return
+        elif not self._hydrate_lazily:
+            raise ExecutionError(
+                "Object has not been hydrated and doesn't support lazy hydration."
+                " This might happen if an object is defined on a different stub,"
+                " or if it's on the same stub but it didn't get created because it was"
+                " in a different module."
```

Comment:
I don't think it's immediately clear what "different module" means here. Wdyt of explicitly mentioning "if it's not imported in global scope"?

Reply:
sure